lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -604,6 +604,47 @@ translate_remote(maps, merge, Meta, [Map1, Map2], S) ->
     {[TMap1, TMap2], TS} ->
       {{call, Ann, {remote, Ann, {atom, Ann, maps}, {atom, Ann, merge}}, [TMap1, TMap2]}, TS}
   end;
+translate_remote(Left, Right, Meta, Args, S) when
+    % try to inline some pure functions
+    Left == 'Elixir.String', Right == length;
+    Left == 'Elixir.String', Right == graphemes;
+    Left == 'Elixir.String', Right == codepoints;
+    Left == 'Elixir.String', Right == split;
+    Left == 'Elixir.MapSet', Right == new;
+    Left == 'Elixir.Duration', Right == 'new!';
+    Left == 'Elixir.URI', Right == new;
+    Left == 'Elixir.URI', Right == 'new!';
+    Left == 'Elixir.URI', Right == parse;
+    Left == 'Elixir.URI', Right == encode_query;
+    Left == 'Elixir.URI', Right == encode_www_form;
+    Left == 'Elixir.URI', Right == decode;
+    Left == 'Elixir.URI', Right == decode_www_form
```

Comment:
I would convert this into a separate function. pattern matching is usually more efficient than guards anyway. :)